libraries/Microsoft.Bot.Builder.Ai/LocaleConverter.cs

```diff
@@ -0,0 +1,191 @@
+        }
+        
+        /// <summary>
+        /// Check if a specific locale is available
+        /// </summary>
+        /// <param name="locale"></param>
+        /// <returns></returns>
+        public bool IsLocaleAvailable(string locale)
+        {
+            return mapLocaleToFunction.ContainsKey(locale);
+        }
+        
+        /// <summary>
+        /// Extract date and time from a sentence using Microsoft Recognizer
+        /// </summary>
+        /// <param name="message"></param>
+        /// <param name="fromLocale">Source Locale</param>
+        /// <returns></returns>
+        private List<TextAndDateTime> extractDate(string message, string fromLocale)
+        {
+            List<TextAndDateTime> fndDates = new List<TextAndDateTime>();
+            var culture = Culture.English; 
+            if (fromLocale.StartsWith("fr"))
+            {
+                culture = Culture.French;
+            }
+            else if (fromLocale.StartsWith("de"))
+            {
+                culture = Culture.German;
+            }
+            else if (fromLocale.StartsWith("pt"))
+            {
+                culture = Culture.Portuguese;
+            }
+            else if (fromLocale.StartsWith("zh"))
+            {
+                culture = Culture.Chinese;
+            }
+            else if (fromLocale.StartsWith("es"))
+            {
+                culture = Culture.Spanish;
+            }
+            else if(!fromLocale.StartsWith("en"))
+            {
+                throw (new ArgumentException("Unsupported From Locale"));
+            }
+            var model = DateTimeRecognizer.GetInstance().GetDateTimeModel(culture);
+            var results = model.Parse(message);
+            foreach (ModelResult result in results)
+            {
+                var resolutionValues = (IList<Dictionary<string, string>>)result.Resolution["values"];
+                string type = result.TypeName.Split('.').Last();
+                DateTime moment = new DateTime();
+                if (type.Contains("date") && !type.Contains("range"))
+                {
+                    moment = resolutionValues.Select(v => DateTime.Parse(v["value"])).FirstOrDefault();
+
+                }
+                else if (type.Contains("date") && type.Contains("range"))
+                {
+                    moment = DateTime.Parse(resolutionValues.First()["start"]);
+                }
+                else
+                {
+                    continue;
+                }
+                var curDateTimeText = new TextAndDateTime
+                {
+                    DateTimeObject = moment,
+                    Text = result.Text
+                };
+                fndDates.Add(curDateTimeText);
+            }
+            return fndDates;
+        }
+        
+        /// <summary>
+        /// Convert a message from locale to another locale
+        /// </summary>
+        /// <param name="message"></param>
```

Comment:
Descriptions?

Reply:
Added a description to message parameter